File: Lua/Example_UserInputAndValidation/addTwoNums.lua
--Get two numbers from user and add together. print sum
num1 = nil
num2 = nil

while num1 == nil or not tonumber(num1) or num1 == "" do
  print("Please give me the first number: ")
  num1 = io.read()
end

while num2 == nil or not tonumber(num2) or num2 == "" do
  print("Please give me a second number: ")
  num2 = io.read()
end

print (num1 + num2) 

File: Python/Example_InputandValidation/AddTwoNums.py
#add two numbers from user and return sum

def IsNum(userInput):
  if userInput.isdigit() or isinstance(userInput, float):
    return True
  else:
    return False


def TryAgain(userInput):
  while not IsNum(userInput):
    userInput = input("That's not a number I can add, try again: ")

  return userInput
  
    

num1 = input("Give me a Number: ")
num1 = TryAgain(num1)
num2 = input("Give me Another Number: ")
num2 = TryAgain(num2)
print(float(num1) + float(num2))